test/library/draft/Vector/Vector.chpl

```diff
@@ -0,0 +1,1263 @@
+
+        if end >= 0 && !_withinBounds(end) then
+          boundsCheckHalt("End" + msg + end:string);
+
+        if !_withinBounds(start) then
+          boundsCheckHalt("Start" + msg + start:string);
+      }
+
+      param error = -1;
+
+      if end >= 0 && end < start then
+        return error;
+
+      var result = error;
+
+      on this {
+        _enter();
+
+        const stop = if end < 0 then _size-1 else end;
+
+        for i in start..stop do
+          if x == _data[i] {
+            result = i;
+            break;
+          }
+
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Return the number of times a given element is found in this vector.
+
+      :arg x: An element to count.
+      :type x: `eltType`
+
+      :return: The number of times a given element is found in this vector.
+      :rtype: `int`
+    */
+    proc const count(x: eltType): int {
+      var result = 0;
+
+      on this {
+        _enter();
+
+        var count = 0;
+
+        for item in this do
+          if x == item then
+            count += 1;
+
+        result = count;
+
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Sort the items of this vector in place using a comparator. If no comparator
+      is provided, sort this vector using the default sort order of its elements.
+
+      .. warning::
+
+        Sorting the elements of this vector may invalidate existing references
+        to the elements contained in this vector.
+
+      :arg comparator: A comparator used to sort this vector.
+    */
+    proc ref sort(comparator: ?rec=Sort.defaultComparator) {
+      on this {
+        _enter();
+
+        //
+        // TODO: This is not ideal, but the Sort API needs to be adjusted
+        // before we can sort over vectors directly.
+        //
+        if _size > 1 {
+
+          // Copy current vector contents into an array.
+          var arr: [0..#_size] eltType;
+          for i in 0..#_size do
+            arr[i] = this[i];
+
+          Sort.sort(arr, comparator);
+
+          // This is equivalent to the clear routine.
+          _size = 0;
+          _capacity = _initialCapacity;
+
+          _requestCapacity(arr.size);
+          _extendGeneric(arr);
+          _maybeDecreaseCapacity();
+        }
+        
+        _leave();
+      }
+      return;
+    }
+
+    pragma "no doc"
+    proc _requestCapacity(newCap: int) {
+      if (_capacity >= newCap) then return;
+      if (_capacity == 0) {
+        _capacity = _initialCapacity;
+      }
+      while (_capacity < newCap) {
+        _capacity *= 2;
+      }
+
+      _domain = {0..#_capacity};
+    }
+
+    pragma "no doc"
+    proc _maybeDecreaseCapacity() {
+
+      const threshold = _capacity/2;
+
+      if threshold <= _initialCapacity then
+        return;
+
+      if _size > threshold then
+        return;
+
+      _capacity /= 2; 
+      _domain = {0..#_capacity};
+    }
+    /*
```

Comment:
We try to keep one line gap between each method.

Reply:
https://github.com/chapel-lang/chapel/pull/16048/commits/f17be557993b5d9096beea3a63579fdadb7cda24